src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -262,7 +274,13 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
         Collections.<Modifier>emptySet(),
         Collections.<Modifier>emptySet(),
         Collections.<Modifier>emptySet(),
-        Collections.singleton(Modifier.STATIC));
+        Collections.singleton(Modifier.STATIC)),
+
+    ANNOTATION(
```

Comment:
I think it isn't necessary to expose `Kind` in public. `Kind` and `isAnnotationDeclaration()` are both in `TypeSpec`.